### PR TITLE
Lightning LSP version update / add integration test suite for lightning and lwc

### PR DESCRIPTION
**Pull request by nrkruk**
### What does this PR do?
- Added integration tests for Lightning and LWC packages
- Update the LSP versions

### CODE REVIEW

File: scripts/download-vscode.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    let vscodeBasePath = path.dirname(path.dirname(vscodeFullPath));
+
+    console.log('Directory Name: ' + vscodeDirname);
+    console.log('Base Path: ' + vscodeBasePath);
+    console.log('Full Path: ' + vscodeFullPath);
+
+    // TODO make this configurable as per package or global?
+
+    // For each extension, copy over the vscode binary
+    for (let i = 0; i < extensionDirectories.length; i++) {
```
Comment (lcampos) [inline scripts/download-vscode.js]: I know that this resembles how running integration test work right now but did we look at just downloading vscode at a top level location and then just use that vscode to run all the tests ?

File: scripts/run-vscode-integration-tests-with-top-level-extensions.js [hunk 2]
```diff
@@ -8,23 +8,24 @@ const path = require('path');
...
-    cwd,
-    'out',
-    'test',
-    'vscode-integration'
-  )}' node ./node_modules/vscode/bin/test`
-);
+  );
+  const CODE_TESTS_PATH = path.join(cwd, 'out', 'test', 'vscode-integration');
+
+  const CODE_VERSION = '1.41.1';
```
Comment (lcampos) [inline scripts/run-vscode-integration-tests-with-top-level-extensions.js]: Why are we hardcoding the version of vscode to download and run tests with ? Right now we always download the latest stable version. In the past we've hardcoded the vscode version only when the stable version has regressions that cause the tests to fail and remove it once vscode addresses the issue.

Comment (nrkruk) [reply scripts/run-vscode-integration-tests-with-top-level-extensions.js]: yeah were working through how to fix this still -- Its temporary but good catch

File: scripts/download-vscode.js (hunk 1 above)
Comment (nrkruk) [reply scripts/download-vscode.js]: I didn't try having them all point to the top level directory, but this is better than what currently happens where we download the binary and extract it in every package. Now it just downloads it once and extracts it everywhere

File: packages/salesforcedx-vscode-lightning/test/vscode-integration/telemetry/telemetry.test.ts [hunk 3]
```diff
@@ -1,71 +1,126 @@
...
-    await telemetryService.sendExtensionDeactivationEvent();
-    assert.calledOnce(sendEvent);
-
-    const expectedData = {
-      extensionName: 'salesforcedx-vscode-lightning'
-    };
-    assert.calledWith(sendEvent, 'deactivationEvent', expectedData);
-  });
-});
+// /*
```
Comment (nrkruk) [inline packages/salesforcedx-vscode-lightning/test/vscode-integration/telemetry/telemetry.test.ts]: TODO - this doesn't need commented out any longer as its fixed in develop

File: scripts/download-vscode.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    console.log('Executable Path: ' + executablePath);
+
+    let vscodeIndex = executablePath.indexOf('.vscode-test') + 13;
+
+    // 'vscode-1.41.1'
+    let vscodeDirname = executablePath.substring(
+      vscodeIndex,
+      executablePath.indexOf(path.sep, vscodeIndex)
+    );
+    process.env.CODE_VERSION = vscodeDirname;
```
Comment (nrkruk) [inline scripts/download-vscode.js]: This doesn't work, we'll have to check the copied directory name to get the version

File: scripts/download-vscode.js (hunk 1 above)
Comment (lcampos) [reply scripts/download-vscode.js]: Ok, let's log a work item to try that approach later. No need to change it as part of this PR.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { assert, expect } from 'chai';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+let doc: vscode.TextDocument;
+let editor: vscode.TextEditor;
```
Comment (bpbuch) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts]: `doc` and `editor` don't need to be defined in the global scope.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+    }
+
+    lwcDir = path.join(
+      vscode.workspace.workspaceFolders![0].uri.fsPath,
+      'force-app',
+      'main',
+      'default',
+      'lwc'
+    );
+    await coreExtension.activate();
```
Comment (bpbuch) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts]: We should try to run the tests without explicitly activating the extensions.

File: packages/salesforcedx-vscode-lightning/test/vscode-integration/intellisense.test.ts [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+  /**
+   * Test that aura markup intellisense contains aura, lightning, custom aura, custom lwc tags
+   */
+
+  it('Aura markup intellisense', async function() {
+    const docUri = vscode.Uri.file(
+      path.join(auraDir, 'DemoComponent', 'DemoComponent.cmp')
+    );
+    const doc = await vscode.workspace.openTextDocument(docUri);
+    const editor = await vscode.window.showTextDocument(doc);
```
Comment (bpbuch) [inline packages/salesforcedx-vscode-lightning/test/vscode-integration/intellisense.test.ts]: Nit: doc and editor could be initialized in beforeEach.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+ */
+
+import { assert, expect } from 'chai';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+describe('LWC Intellisense Test Suite', function() {
+  let lwcDir: string;
+
+  before(async function() {
```
Comment (bpbuch) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts]: This doesn't need to be async.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+          { label: 'c-hello-binding', kind: vscode.CompletionItemKind.Property }
+        ]
+      });
+    } catch (error) {
+      throw error;
+    }
+  });
+
+  it('LWC JS @Salesforce Import Intellisense', async function() {
+    this.timeout(10000);
```
Comment (bpbuch) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts]: A comment is needed to explain why the timeout is increased.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/jsconfig.test.ts [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+const TEST_COMPONENT_NAME = 'testComponent';
+const CREATE_COMPONENT_NAME = 'createComponent';
+
+describe('jsconfig Test Suite', function() {
+  let lwcDir: string;
+
+  let configPath: string;
+  let config: object;
+
+  before(async function() {
```
Comment (bpbuch) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/jsconfig.test.ts]: This function doesn't need to be async.

**Pull request merged**